src/components/ImportForm/GitImportForm.tsx

```diff
@@ -110,7 +115,7 @@ const GitImportForm: React.FunctionComponent<React.PropsWithChildren<GitImportFo
         // no. of secrets with dry run requests
         (values?.importSecrets.length ?? 0) * 2;
       setProgressBarData({ maxValueForProgressBar, noOfComponents });
-      return createResources(values, ImportStrategy.GIT, workspace)
+      createResources(values, ImportStrategy.GIT, workspace)
```

Comment:
If you don't return the promise to formik it will not set proper state values like isSubmitting.

Reply:
@sahil143 ^^

Reply:
We manually set the submitting state in case of an error and if there is no error it will redirect. There is a delay in opening the custom pipeline modal during which the create application button is enabled because promise is resolved and formik is setting isSumbitting to true. To prevent this behaviour I removed the return statement.